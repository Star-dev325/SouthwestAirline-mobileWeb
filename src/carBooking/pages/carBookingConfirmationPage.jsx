// @flow
import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import PageHeader from 'src/shared/components/pageHeader';
import HideForWebView from 'src/shared/components/hideForWebView';
import WithHideLoginButton from 'src/shared/enhancers/withHideLoginButton';
import Container from 'src/shared/components/container';
import Message from 'src/shared/components/message';
import CarBookingDriverCard from 'src/carBooking/components/carBookingDriverCard';
import CarBookingTotalPrice from 'src/carBooking/components/carBookingTotalPrice';
import ConfirmFooter from 'src/carBooking/components/confirmFooter';
import * as WebViewActions from 'src/shared/actions/webViewActions';
import i18n from '@swa-ui/locale';

import type { CarReservationType } from 'src/carBooking/flow-typed/carBooking.types';

type Props = {
  carReservation: CarReservationType,
  bookingResponse: {
    confirmationNumber: string,
    driver: {
      firstName: string,
      lastName: string
    }
  },
  isWebView?: boolean,
  history: *,
  enableNavigationControlsFn: (boolean) => void,
  displayAppReviewFn: () => void
};

export class CarBookingConfirmationPage extends React.Component<Props> {
  componentDidMount() {
    const { isWebView, enableNavigationControlsFn, displayAppReviewFn } = this.props;

    if (isWebView) {
      enableNavigationControlsFn(false);
      displayAppReviewFn();
    }
  }

  render() {
    const {
      carReservation,
      bookingResponse: { confirmationNumber, driver },
      isWebView,
      history: { push }
    } = this.props;

    return (
      <div className="car-booking-confirmation">
        <PageHeader noBottomPadding>
          {i18n('CAR_BOOKING__PURCHASE_CONFIRMATION__HEADER')}
          {isWebView && (
            <a
              className="right white regular page-header--right-button"
              onClick={() => {
                push('/');
              }}
            >
              {i18n('CAR_BOOKING__DONE')}
            </a>
          )}
        </PageHeader>

        <Container inverted>
          <Message status="success" className="p0 small">
            <h3 className="xxlarge bold white inline-block">{i18n('CAR_BOOKING__PURCHASE_CONFIRMATION__TITLE')}</h3>
          </Message>
        </Container>

        <CarBookingDriverCard carReservation={carReservation} confirmationNumber={confirmationNumber} driver={driver} />

        <CarBookingTotalPrice carReservation={carReservation} />

        <HideForWebView>
          <ConfirmFooter />
        </HideForWebView>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  carReservation: _.get(state, 'app.carBooking.carPricingPage.carReservation'),
  bookingResponse: _.get(state, 'app.carBooking.carBookingConfirmationPage.response'),
  isWebView: _.get(state, 'app.webView.isWebView')
});

const mapDispatchToProps = {
  enableNavigationControlsFn: WebViewActions.enableNavigationControls,
  displayAppReviewFn: WebViewActions.displayAppReview
};

const enhancers = _.flowRight(WithHideLoginButton, connect(mapStateToProps, mapDispatchToProps));

export default enhancers(CarBookingConfirmationPage);
