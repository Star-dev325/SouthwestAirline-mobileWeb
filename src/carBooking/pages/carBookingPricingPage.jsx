// @flow
import i18n from '@swa-ui/locale';
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import * as CarBookingActions from 'src/carBooking/actions/carBookingActions';
import CarExtrasForm from 'src/carBooking/components/carExtrasForm';
import type { CarExtraProductType, CarReservationType } from 'src/carBooking/flow-typed/carBooking.types';
import { mapSelectedCarExtras } from 'src/carBooking/transformers/carReservationTransformer';
import CarReservation from 'src/shared/components/carReservation';
import PageHeader from 'src/shared/components/pageHeader';
import { CAR_BOOKING_PRICING_FORM } from 'src/shared/constants/formIds';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import WithShowOnlyLoginButton from 'src/shared/enhancers/withShowOnlyLoginButton';
import type { Push } from 'src/shared/flow-typed/shared.types';
import type { FormData } from 'src/shared/form/flow-typed/form.types';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

type Props = {
  carReservation: CarReservationType,
  carExtras: Array<CarExtraProductType>,
  productId: string,
  saveSelectedExtrasFn: (Array<string>) => void,
  saveCarReservationFn: (CarReservationType) => void,
  loadUserAccountInfoFn: (string) => void,
  isUserLoggedIn: boolean,
  push: Push,
  saveUserAccountDriverInfoFn: (*) => void,
  saveUserAccountContactInfoFn: (*) => void,
  isWebView?: boolean
};

export class CarBookingPricingPage extends React.Component<Props> {
  _saveSelectedExtras = (formData: FormData) => {
    const {
      saveSelectedExtrasFn,
      saveCarReservationFn,
      loadUserAccountInfoFn,
      isUserLoggedIn,
      push,
      saveUserAccountDriverInfoFn,
      saveUserAccountContactInfoFn,
      carReservation,
      carExtras
    } = this.props;

    const selectedCarExtras = _.compact(
      _.map(formData, (value, key) => {
        if (value) {
          return key;
        }
      })
    );
    const purchaseNormalizedRoute = getNormalizedRoute({ routeName: 'purchase' });

    saveSelectedExtrasFn(selectedCarExtras);
    saveCarReservationFn({
      carReservationItinerary: carReservation.carReservationItinerary,
      carReservationDetail: {
        ...carReservation.carReservationDetail,
        selectedCarExtras: mapSelectedCarExtras(carExtras, selectedCarExtras)
      }
    });

    if (isUserLoggedIn) {
      loadUserAccountInfoFn(purchaseNormalizedRoute);
    } else {
      saveUserAccountDriverInfoFn({});
      saveUserAccountContactInfoFn({});
      push(purchaseNormalizedRoute);
    }
  };

  render() {
    const { carExtras, productId, carReservation, isWebView, push } = this.props;

    return (
      <div>
        <PageHeader>{i18n('CAR_BOOKING__PRICE')}</PageHeader>
        <CarReservation {...carReservation} />
        <CarExtrasForm
          formId={CAR_BOOKING_PRICING_FORM}
          productId={productId}
          className="p4 car-extras-form-content"
          carExtras={carExtras}
          carReservation={carReservation}
          onSubmit={this._saveSelectedExtras}
          isWebView={isWebView}
          push={push}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  carReservation: _.omit(
    _.get(state, 'app.carBooking.carPricingPage.carReservation'),
    'carReservationDetail.selectedCarExtras'
  ),
  productId: _.get(state, 'app.carBooking.carPricingPage.response.productId'),
  carExtras: _.get(state, 'app.carBooking.carPricingPage.response.extras', []),
  isUserLoggedIn: _.get(state, 'app.account.isLoggedIn', false),
  isWebView: _.get(state, 'app.webView.isWebView')
});

const mapDispatchToProps = {
  loadUserAccountInfoFn: CarBookingActions.loadUserAccountInfo,
  saveSelectedExtrasFn: CarBookingActions.saveSelectedExtras,
  saveCarReservationFn: CarBookingActions.saveCarReservation,
  saveUserAccountDriverInfoFn: CarBookingActions.saveUserAccountDriverInfo,
  saveUserAccountContactInfoFn: CarBookingActions.saveUserAccountContactInfo
};

const enhancers = _.flowRight(
  withBodyClass('car-booking_price'),
  WithShowOnlyLoginButton,
  withConnectedReactRouter,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(CarBookingPricingPage);
