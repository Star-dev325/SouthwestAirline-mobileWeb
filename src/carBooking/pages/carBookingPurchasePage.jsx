// @flow
import i18n from '@swa-ui/locale';
import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import LoginBanner from 'src/airBooking/components/loginBanner';
import * as CarBookingActions from 'src/carBooking/actions/carBookingActions';
import CarBookingPurchaseForm from 'src/carBooking/components/carBookingPurchaseForm';
import type {
  CarBookingContactInfoType, CarResultVehicleType, DriverInfoType
} from 'src/carBooking/flow-typed/carBooking.types';
import TransformReservationCarRequestToApi from 'src/carBooking/transformers/reservationCarRequestTransformer';
import * as WebViewActions from 'src/shared/actions/webViewActions';
import PageHeader from 'src/shared/components/pageHeader';
import { CAR_BOOKING_PURCHASE_FORM } from 'src/shared/constants/formIds';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import withHideLoginButton from 'src/shared/enhancers/withHideLoginButton';
import type { Push, CurrencyType } from 'src/shared/flow-typed/shared.types';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

type Props = {
  driverInfo: ?DriverInfoType,
  contactInfo: CarBookingContactInfoType,
  totalWithTaxesAndCurrencyCode: CurrencyType,
  reserveCarFn: (*, boolean) => void,
  showNativeAppLoginFn: () => void,
  isUserLoggedIn: boolean,
  isWebView: boolean,
  selectedCarResult: CarResultVehicleType,
  selectedExtras: Array<string>,
  push: Push
};

const contactFields = ['driverIsoCountryCode', 'driverPhoneNumber', 'confirmationEmail'];

export class CarBookingPurchasePage extends React.Component<Props> {
  _reserverCar = (formData: FormData) => {
    const { reserveCarFn, isUserLoggedIn, selectedCarResult, selectedExtras, driverInfo } = this.props;
    const requestData = _.merge({}, driverInfo, formData);
    const request = TransformReservationCarRequestToApi(requestData, selectedCarResult, selectedExtras);

    reserveCarFn(request, isUserLoggedIn);
  };

  _goToDriverInfoEdit = () => {
    this.props.push(getNormalizedRoute({ routeName: 'driverInfo' }));
  };

  render() {
    const { totalWithTaxesAndCurrencyCode, driverInfo, contactInfo, isUserLoggedIn, isWebView, showNativeAppLoginFn } =
      this.props;

    return (
      <div>
        <PageHeader>{i18n('CAR_BOOKING__PURCHASE')}</PageHeader>

        {isWebView && !isUserLoggedIn && <LoginBanner onClick={showNativeAppLoginFn} />}

        <CarBookingPurchaseForm
          formId={CAR_BOOKING_PURCHASE_FORM}
          isUserLoggedIn={isUserLoggedIn}
          driverInfo={isUserLoggedIn ? driverInfo : null}
          initialFormData={_.pick(contactInfo, contactFields)}
          onSubmit={this._reserverCar}
          totalWithTaxesAndCurrencyCode={totalWithTaxesAndCurrencyCode}
          onDriverInfoClick={this._goToDriverInfoEdit}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  driverInfo: _.get(state, 'app.carBooking.userInfo.driverInfo'),
  contactInfo: _.get(state, 'app.carBooking.userInfo.contactInfo'),
  totalWithTaxesAndCurrencyCode: _.get(
    state,
    'app.carBooking.carPricingPage.carReservation.carReservationDetail.totalWithTaxesAndCurrencyCode'
  ),
  isWebView: _.get(state, 'app.webView.isWebView'),
  isUserLoggedIn: _.get(state, 'app.account.isLoggedIn', false),
  selectedCarResult: _.get(state, 'app.carBooking.carPricingPage.selectedCar'),
  selectedExtras: _.get(state, 'app.carBooking.carPricingPage.selectedExtras')
});

const mapDispatchToProps = {
  reserveCarFn: CarBookingActions.reserveCar,
  showNativeAppLoginFn: WebViewActions.showNativeAppLogin
};

const enhancers = _.flowRight(
  withBodyClass('car-booking_purchase'),
  withHideLoginButton,
  withConnectedReactRouter,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(CarBookingPurchasePage);
