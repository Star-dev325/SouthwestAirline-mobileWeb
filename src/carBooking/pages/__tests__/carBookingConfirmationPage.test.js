import i18n from '@swa-ui/locale';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { CarBookingConfirmationPage } from 'src/carBooking/pages/carBookingConfirmationPage';
import CarReservationBuilder from 'test/builders/model/carReservationBuilder';
import { createMockedFormStore } from 'test/unit/helpers/formTestUtils';

describe('CarBookingConfirmationPage', () => {
  let pushStub;
  let enableNavigationControlsFnStub;
  let displayAppReviewStub;

  beforeEach(() => {
    pushStub = jest.fn();
    enableNavigationControlsFnStub = jest.fn();
    displayAppReviewStub = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('component did mount', () => {
    it('should not call enableNavigationControlsFnStub or displayAppReviewStub when not in web view state', () => {
      const { container } = createComponent({ isWebView: false });

      expect(container).toMatchSnapshot();
    });

    it('should call enableNavigationControlsFnStub and displayAppReviewStub when in web view state', () => {
      const { container } = createComponent({ isWebView: true });

      expect(container).toMatchSnapshot();
    });
  });

  describe('render', () => {
    it('should render page header', () => {
      const { container } = createComponent();

      expect(container.querySelector('.page-header').textContent).toEqual(
        i18n('CAR_BOOKING__PURCHASE_CONFIRMATION__HEADER')
      );
    });

    it('should render container', () => {
      const { container } = createComponent();

      expect(container.querySelector('.label-container--label').textContent).toEqual(
        i18n('CAR_BOOKING__CAR_RESERVATION__PICK_UP')
      );
    });

    it('should render car booking driver card with confirmation number', () => {
      const { container } = createComponent();

      expect(container.querySelector('.mx4 car-booking-driver-card')).toMatchSnapshot();
    });

    it('should render car booking total price', () => {
      const { container } = createComponent();

      expect(container.querySelector('.carbooking-total-price price-total')).toMatchSnapshot();
    });

    it('should render car booking confirmation footer', () => {
      const { container } = createComponent();

      expect(container.querySelector('.confirm-footer mt6')).toMatchSnapshot();
    });
  });

  describe('webview', () => {
    it('should render DONE button', () => {
      const { container } = createComponent({ isWebView: true });

      expect(container.querySelector('.page-header--right-button')).toMatchSnapshot();
    });

    it('should navigate to home page when user clicks DONE button', () => {
      const { container } = createComponent({ isWebView: true });

      fireEvent.click(container.querySelector('.page-header--right-button'));

      expect(pushStub).toHaveBeenCalledWith('/');
    });
  });

  const createComponent = (extraProps = {}) => {
    const defaultProps = {
      carReservation: new CarReservationBuilder().build(),
      bookingResponse: {
        confirmationNumber: 'AJC129876234',
        driver: {
          firstName: 'Amy',
          lastName: 'Awesome'
        }
      },
      history: {
        push: pushStub
      },
      enableNavigationControlsFn: enableNavigationControlsFnStub,
      displayAppReviewFn: displayAppReviewStub
    };

    const newProps = {
      ...defaultProps,
      ...extraProps
    };
   
    return render(
      <BrowserRouter>
        <Provider store={createMockedFormStore()}>
          <CarBookingConfirmationPage {...newProps} />
        </Provider>
      </BrowserRouter>
    );
  };
});
