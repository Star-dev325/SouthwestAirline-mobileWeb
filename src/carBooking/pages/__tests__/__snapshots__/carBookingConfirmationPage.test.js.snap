// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarBookingConfirmationPage component did mount should call enableNavigationControlsFnStub and displayAppReviewStub when in web view state 1`] = `
<div>
  <div
    class="car-booking-confirmation"
  >
    <div>
      <div
        class="page-header"
        id=""
      >
        CAR_BOOKING__PURCHASE_CONFIRMATION__HEADER
        <a
          class="right white regular page-header--right-button"
        >
          CAR_BOOKING__DONE
        </a>
      </div>
      <div />
    </div>
    <div
      class="custom-container bgpdkblue white"
    >
      <div
        class="swa-message p0 small success"
      >
        <span
          class="icon-container"
        >
          <i
            class="icon icon_check"
          />
        </span>
        <span
          class="message--text"
        >
          <h3
            class="xxlarge bold white inline-block"
          >
            CAR_BOOKING__PURCHASE_CONFIRMATION__TITLE
          </h3>
        </span>
      </div>
    </div>
    <div
      class="mx4 car-booking-driver-card"
    >
      <div
        class="bgpdkblue white px5 py3 xlarge bold mt4"
      >
        CAR_BOOKING__PURCHASE_CONFIRMATION__DRIVER
      </div>
      <div
        class="bgwhite pdkblue px5 py2"
      >
        <div
          class="xxlarge py2 bold ellipsis overflow-hidden"
        >
          Amy Awesome
        </div>
        <div
          class="large gray5"
        >
          CAR_BOOKING__PURCHASE_CONFIRMATION__CONFIRMATION_NUMBER
          <span
            class="bold pdkblue"
          >
            AJC129876234
          </span>
        </div>
      </div>
      <div
        class="bgwhite mb4 px4 py4"
      >
        <div
          class="car-reservation-itinerary"
        >
          <div
            class="car-reservation-itinerary--pick-up"
          >
            <div
              class="pick-up--icon"
            >
              <i
                class="white p4 circle table-cell align-middle large icon-wrapper_black icon icon_car"
              />
              <div
                class="line--vertical line--vertical_top"
              />
            </div>
            <div
              class="line-wrapper"
            >
              <div
                class="line--horizontal"
              />
            </div>
            <div
              class="label-container pick-up--date"
            >
              <div
                class="label-container--label"
              >
                CAR_BOOKING__CAR_RESERVATION__PICK_UP
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="flight-time"
                >
                  <div
                    class="flight-time--time"
                  >
                    <span>
                      11:00
                    </span>
                    <span
                      class="time-period"
                    >
                      AM
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="label-container pick-up--location"
            >
              <div
                class="label-container--label"
              >
                Tuesday, Mar 1, 2016
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="airport-info"
                >
                  <div
                    class="airport-info--code"
                  >
                    DAL
                  </div>
                  <div
                    class="airport-info--detail"
                  >
                    Dallas (Love Field)
                    , 
                    TX
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="car-reservation-itinerary--return"
          >
            <div
              class="return--icon"
            >
              <i
                class="white p4 circle table-cell align-middle large icon-wrapper_green icon icon_car"
              />
              <div
                class="line--vertical line--vertical_bottom"
              />
            </div>
            <div
              class="line-wrapper"
            >
              <div
                class="line--horizontal"
              />
            </div>
            <div
              class="label-container return--date"
            >
              <div
                class="label-container--label"
              >
                CAR_BOOKING__CAR_RESERVATION__RETURN
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="flight-time"
                >
                  <div
                    class="flight-time--time"
                  >
                    <span>
                      11:00
                    </span>
                    <span
                      class="time-period"
                    >
                      AM
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="label-container return--location"
            >
              <div
                class="label-container--label"
              >
                Friday, Mar 4, 2016
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="airport-info"
                >
                  <div
                    class="airport-info--code"
                  >
                    DAL
                  </div>
                  <div
                    class="airport-info--detail"
                  >
                    Dallas (Love Field)
                    , 
                    TX
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="car-reservation-detail mt4 pt5 bdt"
        >
          <div>
            <div
              class="flex flex-cross-center"
            >
              <div
                class="flex6"
              >
                <div>
                  <img
                    class="fit car-reservation--vendor-logo"
                    src="/content/mkt/images/car_vendors/Avis_Logo_results.png"
                  />
                </div>
              </div>
              <div
                class="green flex6 align-right large"
              >
                Earn up to 600 points
              </div>
            </div>
            <div
              class="gray5 my2"
            >
              Rental Counter is located outside of the Airport Terminal. Shuttle is provided.
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__CAR_TYPE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              Economy
            </div>
          </div>
          <div
            class="clearfix mt2"
            data-qa="car-booking-extras-container"
          >
            <div
              class="left large pr7"
            >
              Rental Extras
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              <span
                class="block align-right"
              >
                Ski Rack
              </span>
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__DAILY_RATE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              <div>
                <span>
                  3 Days
                </span>
                 
                CAR_BOOKING__CAR_RESERVATION__SYMBOL
                <div
                  class="currency car-reservation-detail--currency"
                >
                  <span
                    class="money-sign"
                  >
                    $
                  </span>
                  <span
                    data-qa="total-amount"
                  >
                    47.00
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__BASE_RATE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              <div
                class="currency car-reservation-detail--currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  141.00
                </span>
              </div>
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__MILEAGE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              Unlimited
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="carbooking-total-price price-total"
    >
      <div
        class="price-total--info"
      >
        <div
          class="price-line bgpblue white pt3 xxlarge"
        >
          <span
            class="price-line--title"
          >
            CAR_BOOKING__PURCHASE_FORM__TOTAL
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  181.75
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="price-total--info-row"
        >
          <div
            class="price-total--info-col price-total--info-col--dollar"
          >
            <p>
              CAR_BOOKING__PURCHASE_FORM__TOTAL_PRICE_NOTE_INCLUDE
            </p>
            <p>
              CAR_BOOKING__PURCHASE_FORM__TOTAL_PRICE_NOTE_EXCLUDE
            </p>
          </div>
          <div
            class="price-total--info-col"
            data-qa="price-breakdown"
          >
            <p>
              CAR_BOOKING__PURCHASE_FORM__SHOW_PRICE_BREAKDOWN
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="confirm-footer mt6"
      >
        <div>
          <a
            class="nav-item-link link-bar"
            href="/my-account"
          >
            My Account
            <i
              class="nav-item-link--icon icon icon_keyboard-arrow-right"
            />
          </a>
          <a
            class="nav-item-link link-bar"
            href="/"
          >
            Home
            <i
              class="nav-item-link--icon icon icon_keyboard-arrow-right"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CarBookingConfirmationPage component did mount should not call enableNavigationControlsFnStub or displayAppReviewStub when not in web view state 1`] = `
<div>
  <div
    class="car-booking-confirmation"
  >
    <div>
      <div
        class="page-header"
        id=""
      >
        CAR_BOOKING__PURCHASE_CONFIRMATION__HEADER
      </div>
      <div />
    </div>
    <div
      class="custom-container bgpdkblue white"
    >
      <div
        class="swa-message p0 small success"
      >
        <span
          class="icon-container"
        >
          <i
            class="icon icon_check"
          />
        </span>
        <span
          class="message--text"
        >
          <h3
            class="xxlarge bold white inline-block"
          >
            CAR_BOOKING__PURCHASE_CONFIRMATION__TITLE
          </h3>
        </span>
      </div>
    </div>
    <div
      class="mx4 car-booking-driver-card"
    >
      <div
        class="bgpdkblue white px5 py3 xlarge bold mt4"
      >
        CAR_BOOKING__PURCHASE_CONFIRMATION__DRIVER
      </div>
      <div
        class="bgwhite pdkblue px5 py2"
      >
        <div
          class="xxlarge py2 bold ellipsis overflow-hidden"
        >
          Amy Awesome
        </div>
        <div
          class="large gray5"
        >
          CAR_BOOKING__PURCHASE_CONFIRMATION__CONFIRMATION_NUMBER
          <span
            class="bold pdkblue"
          >
            AJC129876234
          </span>
        </div>
      </div>
      <div
        class="bgwhite mb4 px4 py4"
      >
        <div
          class="car-reservation-itinerary"
        >
          <div
            class="car-reservation-itinerary--pick-up"
          >
            <div
              class="pick-up--icon"
            >
              <i
                class="white p4 circle table-cell align-middle large icon-wrapper_black icon icon_car"
              />
              <div
                class="line--vertical line--vertical_top"
              />
            </div>
            <div
              class="line-wrapper"
            >
              <div
                class="line--horizontal"
              />
            </div>
            <div
              class="label-container pick-up--date"
            >
              <div
                class="label-container--label"
              >
                CAR_BOOKING__CAR_RESERVATION__PICK_UP
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="flight-time"
                >
                  <div
                    class="flight-time--time"
                  >
                    <span>
                      11:00
                    </span>
                    <span
                      class="time-period"
                    >
                      AM
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="label-container pick-up--location"
            >
              <div
                class="label-container--label"
              >
                Tuesday, Mar 1, 2016
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="airport-info"
                >
                  <div
                    class="airport-info--code"
                  >
                    DAL
                  </div>
                  <div
                    class="airport-info--detail"
                  >
                    Dallas (Love Field)
                    , 
                    TX
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="car-reservation-itinerary--return"
          >
            <div
              class="return--icon"
            >
              <i
                class="white p4 circle table-cell align-middle large icon-wrapper_green icon icon_car"
              />
              <div
                class="line--vertical line--vertical_bottom"
              />
            </div>
            <div
              class="line-wrapper"
            >
              <div
                class="line--horizontal"
              />
            </div>
            <div
              class="label-container return--date"
            >
              <div
                class="label-container--label"
              >
                CAR_BOOKING__CAR_RESERVATION__RETURN
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="flight-time"
                >
                  <div
                    class="flight-time--time"
                  >
                    <span>
                      11:00
                    </span>
                    <span
                      class="time-period"
                    >
                      AM
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="label-container return--location"
            >
              <div
                class="label-container--label"
              >
                Friday, Mar 4, 2016
              </div>
              <div
                class="label-container--content"
              >
                <div
                  class="airport-info"
                >
                  <div
                    class="airport-info--code"
                  >
                    DAL
                  </div>
                  <div
                    class="airport-info--detail"
                  >
                    Dallas (Love Field)
                    , 
                    TX
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="car-reservation-detail mt4 pt5 bdt"
        >
          <div>
            <div
              class="flex flex-cross-center"
            >
              <div
                class="flex6"
              >
                <div>
                  <img
                    class="fit car-reservation--vendor-logo"
                    src="/content/mkt/images/car_vendors/Avis_Logo_results.png"
                  />
                </div>
              </div>
              <div
                class="green flex6 align-right large"
              >
                Earn up to 600 points
              </div>
            </div>
            <div
              class="gray5 my2"
            >
              Rental Counter is located outside of the Airport Terminal. Shuttle is provided.
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__CAR_TYPE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              Economy
            </div>
          </div>
          <div
            class="clearfix mt2"
            data-qa="car-booking-extras-container"
          >
            <div
              class="left large pr7"
            >
              Rental Extras
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              <span
                class="block align-right"
              >
                Ski Rack
              </span>
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__DAILY_RATE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              <div>
                <span>
                  3 Days
                </span>
                 
                CAR_BOOKING__CAR_RESERVATION__SYMBOL
                <div
                  class="currency car-reservation-detail--currency"
                >
                  <span
                    class="money-sign"
                  >
                    $
                  </span>
                  <span
                    data-qa="total-amount"
                  >
                    47.00
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__BASE_RATE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              <div
                class="currency car-reservation-detail--currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  141.00
                </span>
              </div>
            </div>
          </div>
          <div
            class="clearfix mt2"
          >
            <div
              class="left large pr7"
            >
              CAR_BOOKING__CAR_RESERVATION__MILEAGE
            </div>
            <div
              class="gray5 large overflow-hidden align-right"
            >
              Unlimited
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="carbooking-total-price price-total"
    >
      <div
        class="price-total--info"
      >
        <div
          class="price-line bgpblue white pt3 xxlarge"
        >
          <span
            class="price-line--title"
          >
            CAR_BOOKING__PURCHASE_FORM__TOTAL
          </span>
          <div
            class="price-amount"
          >
            <div
              class="price-line--currency_total"
            >
              <div
                class="currency"
              >
                <span
                  class="money-sign"
                >
                  $
                </span>
                <span
                  data-qa="total-amount"
                >
                  181.75
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="price-total--info-row"
        >
          <div
            class="price-total--info-col price-total--info-col--dollar"
          >
            <p>
              CAR_BOOKING__PURCHASE_FORM__TOTAL_PRICE_NOTE_INCLUDE
            </p>
            <p>
              CAR_BOOKING__PURCHASE_FORM__TOTAL_PRICE_NOTE_EXCLUDE
            </p>
          </div>
          <div
            class="price-total--info-col"
            data-qa="price-breakdown"
          >
            <p>
              CAR_BOOKING__PURCHASE_FORM__SHOW_PRICE_BREAKDOWN
            </p>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="confirm-footer mt6"
      >
        <div>
          <a
            class="nav-item-link link-bar"
            href="/my-account"
          >
            My Account
            <i
              class="nav-item-link--icon icon icon_keyboard-arrow-right"
            />
          </a>
          <a
            class="nav-item-link link-bar"
            href="/"
          >
            Home
            <i
              class="nav-item-link--icon icon icon_keyboard-arrow-right"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CarBookingConfirmationPage render should render car booking confirmation footer 1`] = `null`;

exports[`CarBookingConfirmationPage render should render car booking driver card with confirmation number 1`] = `null`;

exports[`CarBookingConfirmationPage render should render car booking total price 1`] = `null`;

exports[`CarBookingConfirmationPage webview should render DONE button 1`] = `
<a
  class="right white regular page-header--right-button"
>
  CAR_BOOKING__DONE
</a>
`;
